File: client/src/js/app/analytics.ts
import { invoke } from "@tauri-apps/api/core";
import type { AnalyticsEvent } from "../bindings/AnalyticsEvent";

export default class Analytics {
    static track(event: AnalyticsEvent, data?: Record<string, string | number>): void {
        invoke("track_event", { event, data: data ?? null }).catch(() => {});
    }
}

File: client/src-tauri/src/commands/analytics.rs
use crate::analytics::AnalyticsService;
use common::structs::{AnalyticsEvent, AnalyticsEventData};
use std::sync::Arc;
use tauri::State;

#[tauri::command]
pub(crate) async fn track_event(
    event: AnalyticsEvent,
    data: Option<AnalyticsEventData>,
    analytics: State<'_, Arc<AnalyticsService>>,
) -> Result<(), String> {
    analytics.track(event, data);
    Ok(())
}
